File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/repo/ModuleFuncRepository.java
package pers.acp.admin.oauth.repo;

import pers.acp.admin.oauth.base.OauthBaseRepository;
import pers.acp.admin.oauth.entity.ModuleFunc;

import java.util.List;

/**
 * @author zhangbin by 2018-1-17 17:46
 * @since JDK 11
 */
public interface ModuleFuncRepository extends OauthBaseRepository<ModuleFunc, String> {

    List<ModuleFunc> findByAppid(String appId);

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/repo/ModuleFuncRepository.kt
package pers.acp.admin.oauth.repo

import pers.acp.admin.common.base.BaseRepository
import pers.acp.admin.oauth.entity.ModuleFunc

/**
 * @author zhangbin by 2018-1-17 17:46
 * @since JDK 11
 */
interface ModuleFuncRepository : BaseRepository<ModuleFunc, String> {

    fun findByAppId(appId: String): MutableList<ModuleFunc>

    fun findByParentIdIn(idList: MutableList<String>): MutableList<ModuleFunc>

    fun deleteByIdInAndCovert(idList: MutableList<String>, covert: Boolean)

}
